lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -60,6 +79,26 @@ def visit_all(nodes)
 
     private
 
+    sig { params(node: T.any(YARP::ConstantWriteNode, YARP::ConstantOrWriteNode)).void }
+    def add_constant_write_node(node)
+      add_constant(node, fully_qualify_name(node.name.to_s))
+    end
+
+    sig do
+      params(node: T.any(
+        YARP::ConstantPathWriteNode,
+        YARP::ConstantPathAndWriteNode,
+        YARP::ConstantPathOrWriteNode,
+        YARP::ConstantPathOperatorWriteNode,
+      )).void
+    end
+    def add_constant_path_write_node(node)
+      return unless node.target.parent.nil? || node.target.parent.is_a?(YARP::ConstantReadNode)
```

Comment:
Let's maintain the comment.
```suggestion
      # ignore variable constants like `var::FOO` or `self.class::FOO`
      return unless node.target.parent.nil? || node.target.parent.is_a?(YARP::ConstantReadNode)
```